zkfacade/src/main/java/com/yahoo/vespa/curator/SingletonManager.java

```diff
@@ -272,13 +273,12 @@ private void doUnregister(SingletonWorker singleton) {
          * If lock is held, or acquired, ping the ZK cluster to extend our deadline.
          */
         private void renewLease() {
-            if (doom.get() == INVALID) {
+            // Witness value to detect if invalidation occurs between here and successful ping.
+            Instant ourDoom = doom.get();
+            if (ourDoom == INVALID) {
                 logger.log(INFO, "Lease invalidated");
-                doom.set(null);
                 return; // Skip to updateStatus, deactivation, and release the lock.
             }
-            // Witness value to detect if invalidation occurs between here and successful ping.
-            Instant ourDoom = doom.get();
```

Comment:
Witness should be taken before we check for `INVALID`, lest we risk taking an `INVALID` witness, which means we _should_ nullify the lock, but we miss that.